### DISCUSSION TITLE
[Possibly Missing Specification?] XML-Documentation of  "delegate*<>" method parameters

### DISCUSSION THREAD

**Discussion by Unknown6656** [dotnet/csharplang#5914] [General] [answered]
Imagine having the following codebase:
```csharp
/* L01 */ static unsafe class C
/* L02 */ {
/* L03 */     static void M<T>(delegate*<T> ptr) { }
/* L04 */ 
/* L05 */     static void M(delegate*<void> ptr) { }
/* L07 */ 
/* L08 */     /// <inheritdoc cref="C.M{T}(delegate*{T})"/>
/* L09 */     static void M() { }
/* L10 */ }
```
When trying to compile this code (see [here on sharplab.io](https://sharplab.io/#v2:C4LgbgpgTglgZgTwDQBMQGoA+ABAjANgAIBXAOwGcBDOCQ7AJkIGEBYAKAG93Ce6C6ALIQCyAHgAqAPgAUKCABsIAc0rAIAKgmTCAB2BQAlIQ6EAvt14WeeItiHDZC5ao2i72vYeNmrhXwHpAwlEYUgALaBhgFAB7AGNCOKgIOABeACImADphDnFTR0UVNXU80wN0/0lfG0ERaSMTczZTIA=)), I receive the following warnings on line `8`:

> warning CS1584: XML comment has syntactically incorrect cref attribute 'C.M{T}(delegate*{T})'
> warning CS1658: ) expected. See also error CS1026.

This error occurs regardless on whether one writes
```csharp
/// <inheritdoc cref="C.M{T}(delegate*{T})"/>
```
or
```csharp
/// <inheritdoc cref="C.M(delegate*{void})"/>
```
This therefore does not seem to be an issue with the generic method type parameters or their usage inside the declaration of the parameter `ptr`.

I am now wondering: Is this an issue inside the C# language specs or inside the Roslyn compiler?

**Comment by Unknown6656** [comment]
I also opened an issue on `roslyn` (https://github.com/dotnet/roslyn/issues/60135) in order to facilitate the cross-repo discussion.

**Comment by 333fred** [comment, marked as answer]
Svick linked you to the duplicate issue: https://github.com/dotnet/roslyn/issues/48363. To be frank, yes this is on my plate, but you are precisely the third person to ask for this in nearly as many years. It just hasn't risen to the top of my priority list, and requires coordination with a few other teams to define it as we need to encode new concepts that can't be expressed currently.

**Reply by Unknown6656** [reply]
Thanks for clearing that one up. I can understand why not a whole lot of people have experienced this issue -- few developers like documenting their code and even fewer expose public APIs which take `delegate*<>`-parameters 😉
